src/main/java/org/elasticsearch/index/query/QueryFilterParser.java

```diff
@@ -19,17 +19,14 @@
 
 package org.elasticsearch.index.query;
 
-import org.apache.lucene.search.Filter;
+import org.apache.lucene.search.ConstantScoreQuery;
 import org.apache.lucene.search.Query;
-import org.apache.lucene.search.QueryWrapperFilter;
 import org.elasticsearch.common.inject.Inject;
 
 import java.io.IOException;
 
-/**
- *
- */
-public class QueryFilterParser implements FilterParser {
+@Deprecated
+public class QueryFilterParser implements QueryParser {
```

Comment:
I see. I am afraid this is going to be a problem in the query-parse-refactoring branch, where we are moving part of the parse method (toQuery) to the builders, to convert an es query to a lucene query. We wouldn't be able to do this though if the builder is gone. Can we instead deprecate the builder too? They should live together, either be removed at the same time or both stay. Also, when we will move parsing to the coordinating node, we will need an intermediate representation of the query, which is the current builder (will become streamable), which we will need for the rest layer too.